### ISSUE TITLE
HTTP events are not triggered

### ISSUE THREAD

**Issue by code-distortion** [cognesy/instructor-php#33]
Hi there.  I've noticed that the HTTP events aren't triggered.  I'm currently using OpenAI and wanted to find out which HTTP requests are being sent.

``` php
$person = (new Instructor())
    ->onEvent(RequestSentToLLM::class, fn($event) => dump($event)) // not triggered
    ->onEvent(ResponseReceivedFromLLM::class, fn($event) => dump($event)) // not triggered
    ->onEvent(RequestToLLMFailed::class, fn($event) => dump($event)) // not triggered
    ->respond(
        messages: 'His name is Jason and he is 28 years old.',
        responseModel: Person::class,
    );
```

I found that changing line 61 in the constructor of `src/Features/LLM/Inference.php` got it to work for me…

From:
``` php
# src/Features/LLM/Inference.php:61
$this->httpClient = $httpClient ?? HttpClient::make($this->config->httpClient);
```

To:
``` php
$this->httpClient = $httpClient ?? HttpClient::make($this->config->httpClient, $this->events);
```

There are several other places that instantiate their own HttpClient if one isn't passed,  so I'm not sure if my suggestion fixes it properly for all situations.

I had a look but wasn't sure of how to mock this up for a new test.

**Comment by ddebowczyk** [comment]
Let me look into it later today. And thanks for reporting.

**Comment by ddebowczyk** [comment]
OK, released v0.10.7 with fixes. Thanks again for catching this.
